src/main/java/org/apache/solr/benchmarks/task/README.md

```diff
@@ -0,0 +1,37 @@
+## Introduction
+This package contains Task (TaskType) implementation that is loaded by class name provided in the solr-bench config.
+
+Take note that most of the common tasks are currently implemented and embedded in the BenchmarkMain.
+
+## Usage
+In the solr-bench config, for example under `task-types`, use `task-by-class` and specify the `task-class`:
+```
+"task-types": {
+    "segment-monitor": {
+        "task-by-class": {
+            "task-class": "org.apache.solr.benchmarks.task.SegmentMonitoringTask",
+            "name": "Monitor segment status in collection",
+            "min-threads": 1,
+            "max-threads": 1,
+            "rpm": 1,
+            "duration-secs" : 600,
+            "params" : {
+                "collection": "solr-bench-test"
+            }
+        }
+    }
+}
+```
+
+
+In the above example, a task of name `segment-monitor` would be created with the implementation of `SegmentMonitoringTask` with params `collection`. Take note the params are specific to the actual task.
+
+Currently, all the config for `task-by-class` task type assumes all fields from `BaseBenchmark` (hence `rpm`, `duration-secs` etc) and would be executed using the `ControlledExecutor` with rate, duration and thread count control.
```

Comment:
Are the `BaseBenchmark` fields required, or are there defaults? Maybe we could make it more clear here if there are required options.

Reply:
thanks for the question!

I actually added flagging to `AbstractTask` to indicate whether it's finite or not. Some tasks could be finite (ie, can be exhausted), for example indexing, while other tasks such as periodic health ping is infinite.

For infinite task, `duration-secs` should be defined other exception will be thrown

I also updated the README a bit. but for dev that wishes to implement the class, it's probably better to follow the actual code.